operator/pkg/controller/virtualip/controller.go

```diff
@@ -157,17 +202,32 @@ func (r *VirtualIPReconciler) Reconcile(ctx context.Context) utils.ReconcilerRes
 
 	changed, err = r.CreateOrUpdateOrDelete(ctx, objsToUpdate, objsToDelete, r.mutate)
 	if err != nil {
+		r.updateCondition(
+			metav1.ConditionUnknown,
+			"ObjectUpdateError",
+			err.Error(),
+		)
 		return utils.Requeue(err)
 	}
 	if changed {
+		r.updateCondition(
+			metav1.ConditionFalse,
+			"ObjectUpdateInProgress",
+			"Creation/Update of various objects in progress",
+		)
 		return utils.EndReconciliation()
 	}
 
 	// TODO: Reconfigure CNI + re-generate the Ingress server certificate
 
+	r.updateCondition(metav1.ConditionTrue, "Configured", "All objects properly configured")
```

Comment:
We stop here? What about checking that the keepalived DaemonSets are actually ready? I don't think it's actually needed atm, and it's even probably not as useful as one might think (we are more interested in knowing the status of the VIPs themselves than that of the Pods), but just wanted to make that clear... It's fine if we say that for now, our status only cares about "K8s objects being properly configured", and once we add monitoring (with our sidecar "API"), we'll add more info about the availability of each VIP.

Reply:
I was wondering about it as well, it was simple to only handle "configuration" but I think I will wait for the DaemonSet pods to be ready as well